packages/rest/src/request-context.ts

```diff
@@ -6,17 +6,90 @@
 import {Context} from '@loopback/context';
 import * as onFinished from 'on-finished';
 import {RestBindings} from './keys';
+import {RestServerResolvedConfig} from './rest.server';
 import {HandlerContext, Request, Response} from './types';
 
 /**
  * A per-request Context combining an IoC container with handler context
  * (request, response, etc.).
  */
 export class RequestContext extends Context implements HandlerContext {
+  /**
+   * Get the protocol used by the client to make the request.
+   * Please note this protocol may be different from what we are observing
+   * at HTTP/TCP level, because reverse proxies like nginx or sidecars like
+   * Envoy are switching between protocols.
+   */
+  get requestedProtocol(): string {
+    return (
+      (this.request.get('x-forwarded-proto') || '').split(',')[0] ||
```

Comment:
Do we wan to check `this.request.get('x-forwarded-proto') || '').split(',')[0]` actually yields `http` or `https`, or do we blindly return whatever was set in the `x-forwarded-proto` header?

Reply:
Good catch.

To be honest, I think the extra complexity of checking whether `x-forwarded-proto` contains http/https is not worth our time right now. Personally, I prefer to wait until there is a user with a real-word scenario where this becomes a problem.

If you feel this edge case is important, then please open a new issue to track it.

This pull request is not modifying LB4's behavior in this particular edge case.

Reply:
I see potential problems in code like `foo[protocol].bar`, where `protocol` was expected to be either `http` or `https`.

I am fine with you suggestion, though.